796.py

```diff
@@ -0,0 +1,18 @@
+class Solution:
+    def rotateString(self, A: str, B: str) -> bool:
+        if len(A) != len(B):
+            return False
+        
+        if len(A) == 0 and len(B) == 0:
+            return True
+        
+        length = len(A)
+        
+        while length > 0:
+            length -= 1
+            A = A[1:len(A)] + A[0:1]
```

Comment:
This part is expensive too. It takes linear time to construct a new string that appends `A[0]` to the end of `A[1:]`.

Also, a pythonic way of writing `A[1:len(A)]` is `A[1:]`.